### PR TITLE
EL-365: Check answers

### PR DESCRIPTION
**Pull request by patrick-laa**
[Link to the Jira ticket](https://dsdmoj.atlassian.net/browse/EL-365)

- Use the existing StepsHelper logic to decide what sections to show on the 'check answers' page
- Use a new presenter, backed by a config file, to populate the contents of each section.
- Make the property and property_entry screens into a single mini-loop like vehicles

Checklist

Before you ask people to review this PR:

    Tests and rubocop should be passing
    Github should not be reporting conflicts; you should have recently run git rebase main.
    There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
    The PR description should say what you changed and why, with a link to the JIRA story.
    You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
    You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/presenters/check_answers_presenter.rb
Comment (starswan) [inline app/presenters/check_answers_presenter.rb:61]: I think the only comment I have is that I'm worried about steps_list_for(x0 - other than that, looks really neat.

File: app/presenters/check_answers_presenter.rb
Comment (starswan) [inline app/presenters/check_answers_presenter.rb:31]: We could maybe make this a bit tighter by having two Structs (with keyword_init: true) rather than OpenStruct. OS upsets me as it responds to all methods with nil.

**Comment by patrick-laa** [comment]
Adjusted as per discussion with @starswan to avoid calling `steps_list_for`, which is really an implementation internal of the current StepsHelper logic.

**Review by starswan — APPROVED** [review]

**Comment by patrick-laa** [comment]
Following UAT we:

- Added the missing date box
- Used HTML-escaped % signs
- Allow the property mortgage field to link back to the property screen if the user has no mortgage

Also rebased to main which means:

- Changes to ensure we passed the now-mandatory slim-lint
- Change to ensure we passed the now-mandatory missing-l10n-key check

I'm also NOT addressing a couple of UAT comments, because the steer from Lel was that the current behaviour is as-intended, namely that on the assets screen, if a user selects that they have no assets, the fields in that section for the possible assets should all display as 'Not applicable' rather than not being displayed at all.

File: config/locales/en.yml
Comment (patrick-laa) [reply config/locales/en.yml:332]: I did a bit of research and [what little I could dig up](https://stackoverflow.com/questions/20290130/how-to-deal-with-semantic-symbols-for-accessibility-in-html) suggested that in general screen readers, if faced with a symbol, will read out the name for that symbol. Which in this case would be the desired behaviour. (It might say "percent owned" or even "percent symbol owned" instead of "percentage owned", but both are accurate representations of the message on-screen.)

I'm happy to be corrected, but in the absence of obvious docs on this case, my instinct would be to wait for a broader content/accessibility review by a subject matter expert. 

In the meantime, I thought the least I could do was use a proper HTML representation of the character in question.

**Pull request merged**
